lib/modules/manager/github-actions/extract.ts

```diff
@@ -17,7 +18,31 @@ const actionRe = regEx(
 const shaRe = regEx(/^(?:[a-f0-9]{40}|[a-f0-9]{64})$/);
 const shaShortRe = regEx(/^[a-f0-9]{6,7}$/);
 
+// detects if we run against a Github Enterprise Server and adds the URL to the beginning of the registryURLs for looking up Actions
+// This reflects the behavior of how GitHub looks up Actions
+// First on the Enterprise Server, then on GitHub.com
+function detectCustomGitHubRegistryUrlsForActions(): PackageDependency {
+  const endpoint = GlobalConfig.get('endpoint');
+  const registryUrls = ['https://github.com'];
+  if (endpoint && GlobalConfig.get('platform') === 'github') {
+    const parsedEndpoint = new URL(endpoint);
+
+    if (
+      parsedEndpoint.host !== 'github.com' &&
+      parsedEndpoint.host !== 'api.github.com'
+    ) {
+      registryUrls.unshift(
+        `${parsedEndpoint.protocol}//${parsedEndpoint.host}`
+      );
+      return { registryUrls };
```

Comment:
I think we should move this to the condition above, we don't want to set the source url If it's the default.

Reply:
If I understand you correctly, that is already the case. 

If it is github.com/api.github.com, it returns an empty object `{}` (which will later default during the lookup to github.com). Only if it is a non-standard host, it will set both the custom registry as well as github.com as secondary host.